controllers/helpers.go

```diff
@@ -582,3 +584,18 @@ func getSubnetClientSideFilter(peUUID string) []*nutanixClient.AdditionalFilter
 func getFilterForName(name string) string {
 	return fmt.Sprintf("name==%s", name)
 }
+
+func hasPEClusterServiceEnabled(peCluster *nutanixClientV3.ClusterIntentResponse, serviceName string) bool {
+	if peCluster.Status == nil ||
+		peCluster.Status.Resources == nil ||
+		peCluster.Status.Resources.Config == nil {
+		return false
+	}
+	serviceList := peCluster.Status.Resources.Config.ServiceList
+	for _, s := range serviceList {
+		if strings.ToUpper(*s) == serviceName {
```

Comment:
don't deref a pointer without nil-checking first else it'll cause the process to panic.